core/common-util/src/main/java/uk/gov/gchq/gaffer/commonutil/ByteArrayEscapeUtils.java

```diff
@@ -53,24 +68,66 @@ private ByteArrayEscapeUtils() {
                 temp[currentPosition++] = b;
             }
         }
-        final byte[] escaped = new byte[currentPosition];
-        System.arraycopy(temp, 0, escaped, 0, currentPosition);
-        return escaped;
+        return currentPosition;
     }
 
     /**
      * Unescapes the provided byte array - this should only be called on byte
      * arrays that have been through the <code>escape</code> method.
      *
-     * @param bytes
-     *            the byte array to unescape
+     * @param bytes the byte array to unescape
      * @return the unescaped byte array
      */
     public static byte[] unEscape(final byte[] bytes) {
-        final byte[] temp = new byte[bytes.length];
+        return unEscapeByLength(bytes, 0, bytes.length);
+    }
+
+    /**
+     * Unescapes the provided byte array - this should only be called on byte
+     * arrays that have been through the <code>escape</code> method.
+     *
+     * @param allBytes The backing byte array which contains the subset to unEscape
+     * @param start    The position to start the unEscape, inclusive.
+     * @param end      The position to end the unEscape, exclusive
+     * @return the unescaped byte array
+     */
+    public static byte[] unEscape(final byte[] allBytes, final int start, final int end) {
+        return unEscapeByPosition(allBytes, start, end);
+    }
+
+
+    /**
+     * Unescapes the provided byte array - this should only be called on byte
+     * arrays that have been through the <code>escape</code> method.
+     *
+     * @param allBytes The backing byte array which contains the subset to unEscape
+     * @param start    The position to start the unEscape, inclusive.
+     * @param end      The position to end the unEscape, exclusive
+     * @return the unescaped byte array
+     */
+    public static byte[] unEscapeByPosition(final byte[] allBytes, final int start, final int end) {
+        return unEscapeByLength(allBytes, start, end - start);
+    }
+
+    /**
+     * Unescapes the provided byte array - this should only be called on byte
+     * arrays that have been through the <code>escape</code> method.
+     *
+     * @param allBytes The backing byte array which contains the subset to unEscape.
+     * @param offset   The position to start the unEscape, inclusive
+     * @param length   The length of bytes to unEscape
+     * @return the unescaped byte array
+     */
+    public static byte[] unEscapeByLength(final byte[] allBytes, final int offset, final int length) {
+        if (allBytes.length < offset + length) {
+            throw new InvalidParameterException(String.format("unEscape parameters larger than allByte.length:%d, offset:%d, length:%d", allBytes.length, offset, length));
```

Comment:
The importance here is the **condition** under which to throw an exception and **type** of exception. Formatting the code to fit within a page width will spread out the number of lines occupied by the creation and throwing of this exception message. This would increase the amount of "code noise" that has to be ignored in order to scan over/read/learn/edit code. 

My vote is to keep exception noise to a min.